### DISCUSSION TITLE
How to do end-to-end test of permissioned deployment?

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#8173] [Q&A] [answered]
@ivanlei asked for some details around extending the docker-based [upgrade-test](https://github.com/Agoric/agoric-sdk/tree/master/packages/deployment/upgrade-test) framework for end-to-end testing of [permissioned deployment](https://docs.agoric.com/guides/coreeval/).

**Comment by dckc** [comment]
## WIP Examples

We had one of these for a "null upgrade" for zoe:

 - [agoric-upgrade-11/zoe-upgrade/zoe-upgrade-driver.sh](https://github.com/Agoric/agoric-sdk/blob/5cc5ec8836dcd0c6e11b10799966b6e74601295d/packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-11/zoe-upgrade/zoe-upgrade-driver.sh)

That evolved into an actual zoe upgrade to add atomic reallocations to zcf, but that test got disabled while we coordinate some things.

I'm working on another example in...

 - https://github.com/Agoric/agoric-sdk/pull/8147

**Comment by dckc** [comment]
@anilhelvaci , as we discussed, [53:kread-start](https://github.com/Agoric/agoric-3-proposals/tree/main/proposals/53%3Akread-start) in [agoric-3-proposals](https://github.com/Agoric/agoric-3-proposals) is a good template now.

Note the docs on [Naming](https://github.com/Agoric/agoric-3-proposals/tree/main#naming):
> Each proposal is defined as a subdirectory of propoals. E.g. 16:upgrade-8. The leading number is its number as submitted to the agoric-3 chain. ... If the proposal is pending and does not yet have a number, use a letter.

**Comment by dckc** [comment]
@turadg and @mhofman are presenting agoric-3-proposals tooling that makes this pretty straightforward.
Here's hoping for a bit of explanation here or linked from here.

**Reply by turadg** [reply]
https://github.com/Agoric/agoric-3-proposals has docs. https://github.com/Agoric/agoric-3-proposals/tree/main/packages/synthetic-chain too

**Comment by dckc** [comment, marked as answer]
Adding an analog to https://github.com/Agoric/agoric-sdk/tree/master/a3p-integration looks like a best practice for MN2 dapps, based on discussion with @mhofman and @Chris-Hibbert 

Here's hoping we find time to elaborate a bit more on this.

cc @Jovonni

**Comment by dckc** [comment]
dapp-offer-up and dapp-agoric-basics have end-to-end deployment testing now.

In some cases, it includes testing all the way out to the web UI.
 - https://github.com/Agoric/dapp-offer-up/blob/main/.github/workflows/pr.yml#L27
 - https://github.com/Agoric/dapp-offer-up/actions/runs/10523555796/job/29158478024

for high level discussion:
 - [Agoric/Synpress: A Custom E2E Testing Framework for Cosmos SDK](https://agoric.com/blog/technology/agoric-synpress-a-custom-e2e-testing-framework-for-cosmos-sdk)

end-to-end testing of dapp-orchestration-basics involves not just one chain running in docker, but several. Landing that is just around the corner:

 - https://github.com/Agoric/dapp-orchestration-basics/pull/32

p.s. completed a few PRs later; for example:

  - https://github.com/Agoric/dapp-orchestration-basics/pull/61
